### PR TITLE
review for G-sgd

### PR DESCRIPTION
**Pull request by MarcoArmenta**

### CODE REVIEW

**Review by jodoin — COMMENTED** [review]
There are rendering error.  Please correct it.

File: article/_posts/2020-09-23-G-SGD.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+The network structure can be regarded as a directed graph $$(\mathcal{O},E)$$, with vertices $$\mathcal{O}=\{ O_1,...,O_{H+d+K} \}$$ and edges $$E$$, where H is the number of hidden neurons and $$|E|=m$$.
+
+A path is represented by a vector $$p=(p_1,...,p_m)^T$$ and if edge e is contained in path p, then $$p_e$=1$ and otherwise $$p_e=0$$. These vectors have L entries that are non-zero. The value of path p is defined as $$v_p(w)= \prod_{i=1}^m w_i^{p_i}$$. The set of all such paths is denoted $$\mathcal{P}$$. Finally, the activation status of path $$p$$ is $$a_p(x;w)=\prod_{j:p_{e_{ij}}=1} \mathbb{I}\big( o_j(x;w) > 0 \big)$$.
```
Comment (lemairecarl) [inline article/_posts/2020-09-23-G-SGD.md:23]: Rendering error, probably because of `$$p_e$=1$`

File: article/_posts/2020-09-23-G-SGD.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+* If $$w=g(w')$$ for some $$g\in \mathcal{G}$$ then $$N_w(x)=N_{w'}(x)$$ for all input vector $$x$$.
+
+![](/article/images/G_SGD/sc02.jpg)
+
+![](/article/images/G_SGD/sc03.jpg)
+
+* Based on this, the authors propose to optimize the values of paths instead of weights.
+
+* They consider the set of path vectors $$\mathcal{P} \subser \{ 0,1 \}^m$$.
```
Comment (lemairecarl) [inline article/_posts/2020-09-23-G-SGD.md:39]: rendering error

File: article/_posts/2020-09-23-G-SGD.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+* Given the structure matrix and the values of basis paths, the values of the weights cannot be determined unless the values of free variables are fixed. 
+
+![](/article/images/G_SGD/sc06.jpg)
+
+![](/article/images/G_SGD/sc07.jpg)
+
+![](/article/images/G_SGD/sc08.jpg)
+
+# $$\mathcal{G}-SGD optimization algorithm$$
```
Comment (lemairecarl) [inline article/_posts/2020-09-23-G-SGD.md:55]: Please put "optimization algorithm" outside the math block

**Review by lemairecarl — CHANGES_REQUESTED** [review]
Please fix the rendering errors.

**Comment by MarcoArmenta** [comment]
How can I see where are the rendering errors? Besides clicking "view file" in the commit, because that does not let me see if there are errors.

**Comment by lemairecarl** [comment]
You will need to preview your post locally, see https://vitalab.github.io/howto/

File: article/_posts/2020-09-23-G-SGD.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+The network structure can be regarded as a directed graph $$(\mathcal{O},E)$$, with vertices $$\mathcal{O}=\{ O_1,...,O_{H+d+K} \}$$ and edges $$E$$, where H is the number of hidden neurons and $$|E|=m$$.
+
+A path is represented by a vector $$p=(p_1,...,p_m)^T$$ and if edge e is contained in path p, then $$p_e=1$$ and otherwise $$p_e=0$$. These vectors have L entries that are non-zero. The value of path p is defined as $$v_p(w)= \prod_{i=1}^m w_i^{p_i}$$. The set of all such paths is denoted $$\mathcal{P}$$. Finally, the activation status of path $$p$$ is $$a_p(x;w)=\prod_{j:p_{e_{ij}}=1} \mathbb{I}\big( o_j(x;w) > 0 \big)$$.
```
Comment (AntoineTheb) [inline article/_posts/2020-09-23-G-SGD.md:23]: edge $$e$$ and path $$p$$ [...] $$L$$ entries

**Review by AntoineTheb — APPROVED** [review]
Minor comments, haven't verified the equations

File: article/_posts/2020-09-23-G-SGD.md (hunk 4 above)
Comment (MarcoArmenta) [reply article/_posts/2020-09-23-G-SGD.md]: Done!

**Pull request merged**